lib/radbeacon/le_scanner.rb

```diff
@@ -7,9 +7,26 @@ def initialize(duration = 5)
       @duration = duration
     end
 
+    def scan_command
+      rout, wout = IO.pipe
+      scan_command_str = "sudo hcitool lescan"
+      pid = Process.spawn(scan_command_str, :out => wout)
+      begin
+        Timeout.timeout(@duration) do
+          Process.wait(pid)
+        end
+      rescue Timeout::Error
+        Process.kill('INT', pid)
```

Comment:
Ok but the process will pretty much always exceed its timeout, `hcitool lescan` is designed to run continuously until the user presses Ctrl-C.  In this case I'm using the timeout every time to end the scan process after the duration is reached so it won't run forever.  That's why I was originally using `SIGINT`, to simulate the user pressing Ctrl-C which is the usual way to end a scan.